src/transformers/models/resnet/modeling_resnet.py

```diff
@@ -416,3 +423,84 @@ def forward(
             return (loss,) + output if loss is not None else output
 
         return ImageClassifierOutputWithNoAttention(loss=loss, logits=logits, hidden_states=outputs.hidden_states)
+
+
+@add_start_docstrings(
+    """
+    ResNet backbone, to be used with frameworks like DETR and MaskFormer.
+    """,
+    RESNET_START_DOCSTRING,
+)
+class ResNetBackbone(ResNetPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+
+        self.resnet = ResNetModel(config)
+
+        # note that this assumes there are always 4 stages
+        self.stage_names = ["stem", "stage1", "stage2", "stage3", "stage4"]
+        self.out_features = config.out_features
+
+        # TODO calculate strides appropriately
+        current_stride = self.resnet.embedder.embedder.convolution.stride[0]
+        self.out_feature_strides = {
+            "stem": current_stride,
+            "stage1": current_stride * 2,
+            "stage2": current_stride * 2,
+            "stage3": current_stride * 2,
+            "stage4": current_stride * 2,
+        }
```

Comment:
cc @amyeroberts @alaradirik it would be nice to get some help here to properly compute the output strides for each of the stages.

Reply:
I've removed the strides for now as they are not necessary for DETR and MaskFormer

Reply:
If we need to add something like this in the future, maybe we can use setattr() similar to the [StyleGAN2](https://github.com/NVlabs/stylegan2-ada-pytorch/blob/main/training/networks.py) implementation to retrieve the stride of each block.